src/main/java/io/kestra/plugin/langchain4j/RAG.java

```diff
@@ -0,0 +1,149 @@
+package io.kestra.plugin.langchain4j;
+
+import dev.langchain4j.rag.content.retriever.ContentRetriever;
+import dev.langchain4j.rag.content.retriever.EmbeddingStoreContentRetriever;
+import dev.langchain4j.service.AiServices;
+import io.kestra.core.models.annotations.Example;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.property.Property;
+import io.kestra.core.models.tasks.RunnableTask;
+import io.kestra.core.models.tasks.Task;
+import io.kestra.core.runners.RunContext;
+import io.kestra.plugin.langchain4j.domain.ChatConfiguration;
+import io.kestra.plugin.langchain4j.domain.EmbeddingStoreProvider;
+import io.kestra.plugin.langchain4j.domain.ModelProvider;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotNull;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+
+import java.util.Optional;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(title = "Retrieval Augmented Generation")
+@Plugin(
+    examples = {
+        @Example(
+            full = true,
+            title = """
+                A Retrieval Augmented Generation (RAG) pipeline that first index documents, then use the RAG task to call an LLM with embedding retrieval.
+                WARNING: the KV embedding store is for quick prototyping only as it store the embedding vectors in a K/V store an load them all in memory.
+                """,
+            code = """
+                id: rag
+                namespace: company.team
+
+                tasks:
+                  - id: ingest
+                    type: io.kestra.plugin.langchain4j.IndexDocument
+                    provider:
+                      type: io.kestra.plugin.langchain4j.gemini.GeminiModelProvider
+                      modelName: gemini-embedding-exp-03-07
+                      apiKey: your_api_key
```

Comment:
Maybe something like this so that in the doc we propagate the best practices to protect secrets:

```suggestion
                      apiKey: '{{ secret(...) }}'
```

Reply:
Examples will be totally rewrtitten by the content team later.